src/fun/builtins.bend

```diff
@@ -175,3 +175,120 @@ hvm to_u24:
 # Casts any native number to an i24.
 hvm to_i24:
   ($([i24] ret) ret)
+
+# Encoding
+
+Utf8/MAX_RUNE = '\u{0010FFFF}'
+Utf8/RUNE_ERROR = '\u{FFFD}'
+#Utf8/RUNE_SELF = 0x80
+#Utf8/MAX = 4
+
+Utf8/t1 = 0b00000000
+Utf8/tx = 0b10000000
+Utf8/t2 = 0b11000000
+Utf8/t3 = 0b11100000
+Utf8/t4 = 0b11110000
+
+Utf8/maskx = 0b00111111
+Utf8/mask2 = 0b00011111
+Utf8/mask3 = 0b00001111
+Utf8/mask4 = 0b00000111
+
+Utf8/rune1max = (- (<< 1 7) 1)
+Utf8/rune2max = (- (<< 1 11) 1)
+Utf8/rune3max = (- (<< 1 16) 1)
+
+Encoding/List/append x []               = [x]
```

Comment:
This is pretty bad because inserting at the end through pattern matching is `O(len)`, making encoding be `O(len^2)`.
Instead we should either write thing in the order they appear and reverse at the end, like this:
```
String/encode_utf8 (String/Nil)       acc = (List/reverse acc)
String/encode_utf8 (String/Cons x xs) acc =
  if (<= x Utf8/rune1max) {
    (String/encode_utf8 xs (List/Cons x acc))
  ...
```
Or use a lambda terminated list like this:
```
let str = @x x
# This appends a character to the end
let str = @x (str (List/Cons char x))
# This appends multiple characters to the end
let str = @x (str (List/Cons char1 (List/Cons char2 (List/Cons char3 x))))
# This appends a character to the start
let str = @x (List/Cons char (str x))
# This finalizes it, converting into a normal list.
let str = (str List/Nil)
```
```